aiapy/calibrate/util.py

```diff
@@ -214,42 +306,70 @@ def get_pointing_table(start, end):
             table[c].unit = "arcsecond / pixel"
         if "INSTROT" in c:
             table[c].unit = "degree"
-    # Remove masking on columns with pointing parameters
     for c in table.colnames:
+        # Remove masking on columns with pointing parameters
         if any(n in c for n in ["X0", "Y0", "IMSCALE", "INSTROT"]) and hasattr(table[c], "mask"):
             table[c] = table[c].filled(np.nan)
     return table
 
 
-def get_error_table(error_table=None):
-    if error_table is None:
-        # This is to work around a parfive bug
-        # https://github.com/Cadair/parfive/issues/121
-        os.environ["PARFIVE_DISABLE_RANGE"] = "1"
-        error_table = fetch_error_table()
-        os.environ.pop("PARFIVE_DISABLE_RANGE")
-    if isinstance(error_table, str | pathlib.Path):
-        table = astropy.io.ascii.read(error_table)
-    elif isinstance(error_table, QTable):
-        table = error_table
+def _fetch_error_table(version: int):
+    # Until the delayed feature from sunpy (v6.1) is out, this function
+    # will need to be like this.
+    @manager.require("error_table_v2", *_URL_HASH_ERROR_TABLE[2])
+    def fetch_error_table_v2():
+        return manager.get("error_table_v2")
+
+    @manager.require("error_table_v3", *_URL_HASH_ERROR_TABLE[3])
+    def fetch_error_table_v3():
+        return manager.get("error_table_v3")
+
+    if version == 2:
+        return fetch_error_table_v2()
+    if version == 3:
+        return fetch_error_table_v3()
+    msg = f"Invalid error table version: {version}, must be 2 or 3"
+    raise ValueError(msg)
+
+
+def get_error_table(source) -> QTable:
+    """
+    Fetches the error table from a SSW mirror or uses a local file if one is
+    provided.
+
+    Parameters
+    ----------
+    source : pathlib.Path, int
+        If input is a pathlib.Path, it is assumed to be a file path to a local error table.
+        If a path is provided as a string, it will error as an invalid source.
+        If the input is an int, it is assumed to be a version of the error table (2, 3).
```

Comment:
My opinion here is the same as with the response table. Using the same argument for a path to a file and a version number is confusing. I'd suggest either using something like "SSW" or `None` and just have that default to the latest version of the error table if a path isn't specified.

Reply:
Replaced.